plugins/inputs/rabbitmq/rabbitmq.go

```diff
@@ -599,20 +686,51 @@ func (r *RabbitMQ) createQueueFilter() error {
 	return nil
 }
 
-func (r *RabbitMQ) shouldGatherExchange(exchange Exchange) bool {
+func (r *RabbitMQ) shouldGatherExchange(exchangeName string) bool {
```

Comment:
This is true, but as far as I can tell the only options are the following:

- Pass in only the name of the exchange (as I've done here)
- Pass in a `Exchange` constructed in `shouldGatherFederationLink` that was created just for that purpose and happens to know that `shouldGatherExchange` only requires a name (since no other properties are set)
- Not filter the gathered federation link data based on the configuration for gathering exchanges (in which case maybe we shouldn't filter for queue-gathering config either?)

None of these seem ideal to me, but if you have a preference (or know of another option) I certainly don't mind taking a different approach.